lib/ansible/module_utils/remote_management/spacewalk.py

```diff
@@ -0,0 +1,333 @@
+                      required=True,
+                      no_log=True,
+                      fallback=(env_fallback, ['SPACEWALK_PASSWORD']),
+                      ),
+        validate_certs=dict(type='bool',
+                            required=False,
+                            default=True,
+                            fallback=(env_fallback, ['SPACEWALK_VALIDATE_CERTS'])),
+    )
+
+
+class Channel(object):
+    def __init__(self, module):
+        self.module = module
+        self.client, self.session = connect_to_api(module)
+        self.channels = {'spacewalk_channels': {}}
+
+    def get_all_channels(self):
+        """ Get all chennels from spacewalk.
+            Fill self.channels for caching pourpose and ready to serves ansible_facts
+        """
+        module = self.module
+        if not self.channels['spacewalk_channels']:
+            channels = self.client.channel.listAllChannels(self.session)
+            for channel in channels:
+                label = channel.pop('label')
+                channel_detail = self.get_channel_detail(label)
+                channel_detail['last_modified'] = str(channel_detail['last_modified'])
+                if 'yumrepo_last_sync' in channel_detail:
+                    channel_detail['yumrepo_last_sync'] = str(channel_detail['yumrepo_last_sync'])
+                channel.update(channel_detail)
+                self.channels['spacewalk_channels'][label] = channel
+
+        ansible_facts = self.channels
+        return ansible_facts
+
+    def get_channel_id(self, channel_label):
+        """ Return integer channel_id passing channel label
+        """
+        if not self.channels['spacewalk_channels']:
+            self.get_all_channels()
+
+        if channel_label in self.channels['spacewalk_channels']:
+            return self.channels['spacewalk_channels'][channel_label]['id']
+
+    def get_channel_detail(self, channel_label):
+        return self.client.channel.software.getDetails(self.session, channel_label)
+
+    def create_channel(self):
+        ch_label = self.module.params['label']
+        ch_name = self.module.params['name']
+        ch_summary = self.module.params['summary']
+        ch_archLabel = self.module.params['archLabel']
+        ch_parentLabel = self.module.params['parentLabel']
+        ch_checksumType = self.module.params['checksumType']
+        ch_gpgurl = self.module.params['gpgurl']
+        ch_gpgid = self.module.params['gpgid']
+        ch_gpgfingerprint = self.module.params['gpgfingerprint']
+        ch_gpgKey = {'url': ch_gpgurl, 'id': ch_gpgid, 'fingerprint': ch_gpgfingerprint}
+
+        try:
+            res = self.client.channel.software.create(self.session, ch_label, ch_name, ch_summary, ch_archLabel, ch_parentLabel, ch_checksumType, ch_gpgKey)
+        except Exception as generic_exception:
+            self.module.fail_json(ch_channel=label, action='Create', msg='Error creating channel: %s' % generic_exception)
```

Comment:
Here you pass the variable `label`. You have never defined it, though (you defined `ch_label` -- maybe you meant that one?). That's the first of the sanity checks which fails.

You might already get that output by running `flake8` on your module.

Reply:
fixed thank you @felixfontein I wrote ch_ prefix in wrong variable